bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/GenericCommunicator.java

```diff
@@ -257,28 +268,40 @@ private void sendPacket(byte[] packet) throws IOException {
         tx.write(packet);
     }
 
-    // TODO think about better approach
     protected byte[] receivePacket() throws InterruptedException, IOException {

```

Comment:
Not completely, but close

I would expect something a little bit different

maybe a field `consecutiveErrors`

```java
protected byte[] receivePacket() throws IOException {
try {
byte[] result = new byte[256];
            rx.read(result);
            ParadoxUtil.printPacket("RX:", result);
            if (result[1] > 0 && result[1] + 16 < 256) {
                return processResponse(Arrays.copyOfRange(result, 0, result[1] + 16));
            }
} catch (...){
}
if (consecturiveErrors < 4) {
 // reschedule receivePacket
resultFuture = scheduler.schedule(receivePacketCallable, 200, TimeUnit.MILLISECONDS);
}

```

Also consider closing / cancelling Future's those from the close method.